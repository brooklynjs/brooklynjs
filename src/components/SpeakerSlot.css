.speaker {
  position: relative;
  padding-left: 30px;
}

.speaker:before {
  background-color: #F7DF1E;
  content: "";
  display: block;
  height: 8px;
  left: 2px;
  position: absolute;
  top: 25px;
  width: 8px;
}

.empty-speaker {
  text-transform: uppercase;
}