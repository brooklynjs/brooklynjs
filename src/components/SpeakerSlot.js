import React from 'react';

import './SpeakerSlot.css';

import {
  speakerSlotName,
  speakerSlotTime
} from '../utils/speakers';

const EmptySpeakerSlot = ({ index }) => {
  const slotName = speakerSlotName(index);

  return (
    <div>
      <li className="ride"></li>

      <li className="speaker">
        <time>{speakerSlotTime(index)}</time>
        <small>Listen to our {slotName} speaker, or better yet</small>
        <div className="empty-speaker">
          <a href="https://github.com/brooklynjs/brooklynjs.github.io/issues/new">
            Be our {slotName} speaker
          </a>
        </div>
      </li>
    </div>
  )
};

const SpeakerSlot = ({ speaker, index }) => {
  if (speaker === null) {
    return (<EmptySpeakerSlot index={index} />)
  }

  return (
    <div>
      <li className="ride"></li>

      <li className="speaker">
        <time>{speakerSlotTime(index)}</time>
        <small>{speaker.title}</small>

        <div>
          <a href={speaker.url}>
            {speaker.name}
          </a>
        </div>
      </li>
    </div>
  )
};

export default SpeakerSlot;