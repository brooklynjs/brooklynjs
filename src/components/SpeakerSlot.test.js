import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import SpeakerSlot from './SpeakerSlot';

describe('an empty speaker slot', () => {
  it('renders', () => {
    const component = (
      <SpeakerSlot speaker={null} index={1} />
    );

    const tree = ReactTestRenderer.create(component).toJSON();

    expect(tree).toMatchSnapshot();
  });
});

describe('a speaker slot with a speaker', () => {
  const speaker = {
    title: 'A Very Cool Talk!',
    name: 'Willman Duffy',
    website: 'https://www.brooklynjs.com'
  };

  it('renders', () => {
    const component = (
      <SpeakerSlot speaker={speaker} index={1} />
    );

    const tree = ReactTestRenderer.create(component).toJSON();

    expect(tree).toMatchSnapshot();
  });
});