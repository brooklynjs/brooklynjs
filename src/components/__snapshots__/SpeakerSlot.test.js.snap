// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`a speaker slot with a speaker renders 1`] = `
<div>
  <li
    className="ride"
  />
  <li
    className="speaker"
  >
    <time>
      8:15p
    </time>
    <small>
      A Very Cool Talk!
    </small>
    <div>
      <a
        href={undefined}
      >
        Willman Duffy
      </a>
    </div>
  </li>
</div>
`;

exports[`an empty speaker slot renders 1`] = `
<div>
  <li
    className="ride"
  />
  <li
    className="speaker"
  >
    <time>
      8:15p
    </time>
    <small>
      Listen to our 
      second
       speaker, or better yet
    </small>
    <div
      className="empty-speaker"
    >
      <a
        href="https://github.com/brooklynjs/brooklynjs.github.io/issues/new"
      >
        Be our 
        second
         speaker
      </a>
    </div>
  </li>
</div>
`;
